htdocs/projet/tasks/time.php

```diff
@@ -589,6 +589,7 @@
 				foreach ($arrayoftasks as $timespent_id => $value) {
 					$userid = $value['user'];
 					//$pu_ht = $value['timespent'] * $fuser->thm;
+					$pu_ht = $fuser->thm;
```

Comment:
The price per hour of the user may change with time.
And total of timespent may have been calculated with different rate that change with time. Each time line may have been entered with a different hourly rate.
Di you try this instead 
$pu_ht = price2num($value['totalvaluetodivideby3600'] / 3600, 'MU');